src/lib/djerba/plugins/fusion/plugin.py

```diff
@@ -63,12 +67,129 @@ def sort_by_actionable_level(row):
         data[core_constants.RESULTS] = results
         data[core_constants.MERGE_INPUTS]['gene_information_merger'] = gene_info
         data[core_constants.MERGE_INPUTS]['treatment_options_merger'] = treatment_opts
         return data
 
+    def process_fusion(self, config, fusion, tsv_file_path, json_template_path, output_dir):
+        base_dir = (directory_finder(self.log_level, self.log_path).get_base_dir())
+        wrapper = self.get_config_wrapper(config)
+        match = re.match(r"(.+)::(.+)", fusion)
+        if match:
+            gene1 = match.group(1)
+            gene2 = match.group(2)
+        else:
+            raise ValueError(f"No valid fusion found for {fusion}.")
+
+        # Initialize breakpoints
+        breakpoint1, breakpoint2 = None, None
+
+        # Open and read the ARRIBA TSV file
+        with open(tsv_file_path, mode='r') as file:
+            reader = csv.DictReader(file, delimiter='\t')
+
+            # Find the correct row based on gene1 and gene2
+            for row in reader:
+                if (row['#gene1'] == gene1 or row['#gene1'] == gene2) and (
+                        row['gene2'] == gene1 or row['gene2'] == gene2):
+                    breakpoint1 = row['breakpoint1']
+                    breakpoint2 = row['breakpoint2']
+                    break
+
+        if not (breakpoint1 and breakpoint2):
+            raise ValueError(f"No matching fusion found in the TSV file for {fusion}.")
+
+        # Function to modify the breakpoint format to "chr:start-end"
+        def format_breakpoint(breakpoint):
+            chrom, pos = breakpoint.split(':')
+            start = int(pos)
+            end = start + 1
+            return f"{chrom}:{start}-{end}"
+
+        formatted_breakpoint1 = format_breakpoint(breakpoint1)
+        formatted_breakpoint2 = format_breakpoint(breakpoint2)
+
+        # Load the JSON template
+        with open(json_template_path, 'r') as json_file:
+            data = json.load(json_file)
+
+        # Update the JSON with the formatted breakpoints
+        data['locus'] = [formatted_breakpoint1, formatted_breakpoint2]
+
+        tumour_id = wrapper.get_my_string(core_constants.TUMOUR_ID)
+        data['tracks'][1]['name'] = tumour_id
+
+        # Search for the BAM and BAI files using glob.glob
+        bam_pattern = f"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bam"
+        bai_pattern = f"/.mounts/labs/prod/whizbam/*/RNASEQ/{tumour_id}.bai"
+
+        bam_files = glob.glob(bam_pattern)
+        bai_files = glob.glob(bai_pattern)
+
+        if bam_files:
+            bam_file = bam_files[0]
+            project = os.path.basename(os.path.dirname(os.path.dirname(bam_file)))
+            filename = os.path.basename(bam_file)
+            data['tracks'][1]['url'] = f"/bams/project/{project}/RNASEQ/file/{filename}"
+        else:
+            raise FileNotFoundError(f"BAM file not found for pattern: {bam_pattern}")
+
+        if bai_files:
```

Comment:
Similar considerations to the BAM files apply here. Also, we should make sure the BAM and BAI files match.

Reply:
Same solution, added project_id to the patter. 
> We should make sure the BAM and BAI files match.
Do you mean something like running samtools idxstats or samtools quickcheck and if those commands fail we throw an error? Because we're looking under the same project directory for files that share the same tumor_id in their filenames

Reply:
No, running samtools would really be overkill -- I just mean ensuring the BAM and BAI paths refer to files with the same filename prefix, in the same directory. As I've previously noted, the code in its current state doesn't guarantee this.

Reply:
The pattern in the code was updated to 
        ```
bam_pattern = f"{core_constants.WHIZBAM_PATTERN_ROOT}/{project_id}/RNASEQ/{tumour_id}.bam"
        bai_pattern = f"{core_constants.WHIZBAM_PATTERN_ROOT}/{project_id}/RNASEQ/{tumour_id}.bai"
```